src/bsoncxx/test/json.cpp

```diff
@@ -70,4 +70,28 @@ TEST_CASE("CXX-941 is resolved") {
     REQUIRE(output ==
             "{ \"obj_name\" : \"{\\\"id1\\\":\\\"val1\\\", \\\"id2\\\":\\\"val2\\\"}\" }");
 }
+
+TEST_CASE("CXX-1246: Legacy Extended JSON") {
+    bsoncxx::types::b_decimal128 decimal_val("-1234E+999");
+    auto doc = make_document(kvp("number", 42), kvp("dec128", decimal_val));
+    std::string output = bsoncxx::to_json(doc.view());
```

Comment:
pretty reasonable to use auto to capture the result of to_json in this test and below.

Reply:
+1